File: thursday/week5/oeyh/10815.py
import sys


def binary_search(array, target, start, end):
    if start > end:
        return 0

    mid = (start + end) //2

    if array[mid] == target:
        return 1
    elif array[mid] > target:
        return binary_search(array, target, start, mid-1)
    else:
        return binary_search(array, target, mid +1, end)


n = int(sys.stdin.readline())
card_arr = list(map(int, sys.stdin.readline().split()))
m = int(sys.stdin.readline())
value_arr = list(map(int, sys.stdin.readline().split()))

card_arr.sort()
result = []

for v in value_arr:
    result.append(binary_search(card_arr, v, 0, n-1))

result = ' '.join(map(str, result))

print(result)

File: saturday/week_15/charmeee/10815.js
let fs = require('fs');
let arr = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
let sangunArr = arr[1].trim().split(' ').map(x=>parseInt(x));
let comArr = arr[3].trim().split(' ').map(x=>parseInt(x));
sangunArr.sort((a,b)=>a-b);

let output =[];
let first = 0;

for( let a in comArr ){
    binary_search(0,sangunArr.length-1,comArr[a]);
}

function binary_search(start,end,target){
    // console.log(`${start} ${end} ${target}`);
    if(start>end){
        output.push(0);
        return;
    }
    let mid = parseInt((start+end)/2);
    // console.log(mid)
    if(target===sangunArr[mid]){
        output.push(1);
        return;
    }
    if(target > sangunArr[mid]){
        return binary_search(mid+1,end,target);
    }else{//target < sangunArr[mid]
        return binary_search(start,mid-1,target);
    }
}

console.log(output.join(' '));